nodeutil/node_map.go

```diff
@@ -66,7 +66,7 @@ func (def *mapAsList) setComparator(c ReflectListComparator) {
 
 func (def *mapAsList) getByKey(r node.ListRequest) (reflect.Value, error) {
 	var empty reflect.Value
-	if r.Key == nil || len(r.Key) == 0 {
+	if r.Key == nil || len(r.Key) == 0 || r.Key[0] == nil {
```

Comment:
we'd really need to check each key for nil.  Maybe write a func and call it in each of the 3 checks

Reply:
done